### PR TITLE
decisionEngine endpoints

### PR DESCRIPTION
**Pull request by Mars-98**
created dict to keep track of key/values, made wallHeight and userHeight endpoint request, fixed route path
https://tim-kaufenberg.atlassian.net/browse/ARC-99

### CODE REVIEW

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # Keys of each element: placement, type
 rocks = []
 
-@route('decisionEngine/shortestPath', method='POST')
+@route('/decisionEngine/shortestPath', method='POST')
```
Comment (coughenberg) [inline decisionengine/decisionengine_endpoint/recieve_rock_info.py:15]: Rip me lol

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py [hunk 2]
```diff
@@ -40,13 +40,31 @@ def find_path():
...
     if not rocks:
         abort(400, 'Bad request')
     for rock in rocks:
         if not (rock.get('placement') or rock.get('type')):
             abort(400, 'Bad request')
 
+    userHeightRequest = request.json.get('userHeight')
+    if userHeightRequest is None:
+        userHeightRequest = '' 
+    elif not isinstance(userHeightRequest, (int, float, complex)):
```
Comment (coughenberg) [inline decisionengine/decisionengine_endpoint/recieve_rock_info.py]: Don't compare vs complex numbers here cause we don't want them

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py [hunk 3]
```diff
@@ -40,13 +40,31 @@ def find_path():
...
+        abort(400, 'Bad user height number input')
+
+    wallHeightRequest = request.json.get('wallHeight')
+    if wallHeightRequest is None:
+        wallHeightRequest = '' 
+    elif not isinstance(wallHeightRequest, (int, float, complex)):
```
Comment (coughenberg) [inline decisionengine/decisionengine_endpoint/recieve_rock_info.py]: Same with the complex type here

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py [hunk 4]
```diff
@@ -40,13 +40,31 @@ def find_path():
...
+        abort(400, 'Bad wall height number input')
+
+    endPts = {
+        "rocks":rocks,
+        "userHeight":userHeightRequest,
+        "wallHeight":wallHeightRequest
+    }
+    print(endPts)
```
Comment (coughenberg) [inline decisionengine/decisionengine_endpoint/recieve_rock_info.py]: Get rid of print statement

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py [hunk 5]
```diff
@@ -40,13 +40,31 @@ def find_path():
...
+
     return {'Message': 'Request successfully recieved, congratz on your rockz', 'Response rocks': rocks}
```
Comment (coughenberg) [inline decisionengine/decisionengine_endpoint/recieve_rock_info.py]: Your dictionary endPts isn't getting returned here, it should be so we know everything is successfull

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py [hunk 6]
```diff
@@ -40,13 +40,31 @@ def find_path():
...
+    wallHeightRequest = request.json.get('wallHeight')
+    if wallHeightRequest is None:
+        wallHeightRequest = '' 
+    elif not isinstance(wallHeightRequest, (int, float, complex)):
+        abort(400, 'Bad wall height number input')
+
+    endPts = {
+        "rocks":rocks,
+        "userHeight":userHeightRequest,
+        "wallHeight":wallHeightRequest
```
Comment (coughenberg) [inline decisionengine/decisionengine_endpoint/recieve_rock_info.py]: I would prefer to not get key-value pairs back that the user did not submit, even if the are empty strings. The client sending the request should only see userHeight and wallHeight if they sent in the values for them. Would you instead only add them to the dictionary if they are in the request?

Also space on the right side of the :

**Review by coughenberg — CHANGES_REQUESTED** [review]

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py (hunk 6 above)
Comment (Mars-98) [reply decisionengine/decisionengine_endpoint/recieve_rock_info.py]: fixed everything, plz check!

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py [hunk 7]
```diff
@@ -49,22 +49,23 @@ def find_path():
...
-    }
-    print(endPts)
-
-    return {'Message': 'Request successfully recieved, congratz on your rockz', 'Response rocks': rocks}
+    endPts = {}
+    add_KV_pair(endPts, 'rocks', rocks)
+    add_KV_pair(endPts, 'userHeight', userHeightRequest)
+    add_KV_pair(endPts, 'wallHeight', wallHeightRequest)
 
+    return endPts
```
Comment (coughenberg) [inline decisionengine/decisionengine_endpoint/recieve_rock_info.py]: The message needs to be kept though, can we layout the response like this?

{ response: { endpoints, message }}

**Review by coughenberg — CHANGES_REQUESTED** [review]

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py [hunk 8]
```diff
@@ -40,13 +40,32 @@ def find_path():
...
+    elif not isinstance(userHeightRequest, (int, float)):
+        abort(400, 'Bad user height number input')
 
+    wallHeightRequest = request.json.get('wallHeight')
+    if wallHeightRequest is None:
+        wallHeightRequest = '' 
+    elif not isinstance(wallHeightRequest, (int, float)):
+        abort(400, 'Bad wall height number input')
+
+    endPts = {}
```
Comment (skitscha) [inline decisionengine/decisionengine_endpoint/recieve_rock_info.py]: Can we rename endPts to something that makes maybe like params?

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py [hunk 9]
```diff
@@ -40,13 +40,32 @@ def find_path():
...
+    if userHeightRequest is None:
+        userHeightRequest = '' 
+    elif not isinstance(userHeightRequest, (int, float)):
+        abort(400, 'Bad user height number input')
 
+    wallHeightRequest = request.json.get('wallHeight')
+    if wallHeightRequest is None:
+        wallHeightRequest = '' 
+    elif not isinstance(wallHeightRequest, (int, float)):
+        abort(400, 'Bad wall height number input')
```
Comment (skitscha) [inline decisionengine/decisionengine_endpoint/recieve_rock_info.py]: Can we move this to its own method that accepts a JSON, error message and the param name that you are looking for?

**Review by coughenberg — APPROVED** [review]

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py (hunk 8 above)
Comment (Mars-98) [reply decisionengine/decisionengine_endpoint/recieve_rock_info.py]: fixed

File: decisionengine/decisionengine_endpoint/recieve_rock_info.py (hunk 9 above)
Comment (Mars-98) [reply decisionengine/decisionengine_endpoint/recieve_rock_info.py]: yup, fixed!

**Review by skitscha — APPROVED** [review]

**Pull request merged**